src/game_battlealgorithm.cpp

```diff
@@ -1004,7 +1004,7 @@ bool Game_BattleAlgorithm::Normal::Execute() {
 			}
 		}
 
-		if (effect > 0) {
+		if ((Player::IsRPG2k() && !Player::IsRPG2kUpdated()) || effect > 0) {
```

Comment:
So it looks like this special case also occurs in the old versions of rm2k3 v1.0.3.0.

So we should change this to `Player::IsLegacy() || effect > 0`